ground/lib/station.js

```diff
@@ -320,10 +321,49 @@ export class Station extends EventEmitter {
         }
       });
 
-      this.radioOut.write(new Buffer(Message.fromPing({ magic })));
+      this.options = { //for sudo
+          cachePassword: true,
+          prompt: 'Password:',
+          spawnOptions: { }
+      };
+
+      let pingListenerDir = path.join(__dirname, '..', '..', 'lib', 'pingListener.py');
+      let pingListener = sudo(['python', '-u', pingListenerDir], this.options); //must execute as root to access /dev/tun0
+       
+      pingListener.stdout.on('data', packet => {
+        if(packet.length === 57) {  //ping packets to ip fe80:.. are exactly 57 bytes. Filter other packets that are received.
+          log.debug('transmitting ping packet: ' + packet.toString('hex'));
+          this.sendMessage(new Buffer(packet));
+          pingListener.kill();
+          resolve();    // Going to change when pong message functionality is added.
+        } 
+      });
+
+      pingListener.stdout.once('data', data => {
+        this.assignIp();
+      });    
+
+    });
+  }
+
+  assignIp() {
+    let ifconfigCmd = ['ifconfig', 'tun0', 'inet6', 'fe80::1234'];       //Assign the tun0 interface an ip address
+    let ifconfig = sudo(ifconfigCmd, this.options); 
+    
+    ifconfig.stdout.once('readable', () => {
+        this.spawnPing();
     });
   }
 
+  spawnPing() {
+    let pingCmd = 'ping6';  
+    let pingCmdArgs = '-c1';     
+    let tunInterface = 'tun0';
+    let balloonIpAddr = 'fe80::aebc:32ff:fe79:57b1'; 
+    let pingArgs = [pingCmdArgs, balloonIpAddr.concat('%', tunInterface)];
+    spawn('ping6', pingArgs);
```

Comment:
does the `ping6` command ever terminate somewhere (either gracefully or forcefully?)